File: PB - More Exercises Python/7.0 Nested Loops - More Exercises/06. Wedding Seats.py
sectors, row_count, seats_count = input(), int(input()), int(input())
even_seats, counter = 0, 0

for sector in range(ord('A'), ord(sectors) + 1):
    row_count += 1
    for row in range(1, row_count):
        if row % 2 == 0:
           even_seats = 2
        else:
            even_seats = 0
        for seats in range(1, seats_count + even_seats + 1):
            print(f'{chr(sector)}{row}{chr(96 + seats)}')
            counter += 1

print(counter)


'''
'B'
3
2
'''

File: PB - More Exercises GO/7.0 Nested Loops - More Exercises/06. Wedding Seats.go
package main

import "fmt"

func main() {
	var sectors string
	var rowCount, seatsCount, evenSeats, counter int
	fmt.Scan(&sectors, &rowCount, &seatsCount)

	for sector := 'A'; sector <= 'Z'; sector++ {
		rowCount++
		for row := 1; row < rowCount; row++ {
			if row%2 == 0 {
				evenSeats = 2
			} else {
				evenSeats = 0
			}

			for seats := 1; seats < (seatsCount + evenSeats + 1); seats++ {
				counter++
				fmt.Printf("%v%v%v\n", string(sector), row, string(96+seats))
			}
		}
		if string(sector) == sectors {
			fmt.Println(counter)
			break
		}
	}

}

/*
name   :06. Wedding Seats
input  :B 3 2
output :
A1a
A1b
A2a
A2b
A2c
A2d
A3a
A3b
B1a
B1b
B2a
B2b
B2c
B2d
B3a
B3b
B4a
B4b
B4c
B4d
20
*/
